Code/Common/include/sitkMemberFunctionFactoryBase.h

```diff
@@ -69,21 +69,16 @@ template <class... TupleArgs>
 struct hash<std::tuple<TupleArgs...>>
 {
 private:
-  // recursive hashing of std::tuple from Sarang Baheti's blog
-  // https://www.variadic.xyz/2018/01/15/hashing-stdpair-and-stdtuple/
   template <size_t Idx, typename... TupleTypes>
-  inline typename std::enable_if<Idx == sizeof...(TupleTypes), void>::type
-  hash_combine_tup(size_t &, const std::tuple<TupleTypes...> &) const
-  {}
-
-  template <size_t Idx, typename... TupleTypes>
-    inline typename std::enable_if < Idx<sizeof...(TupleTypes), void>::type
-                                     hash_combine_tup(size_t & seed, const std::tuple<TupleTypes...> & tup) const
+  void
+  hash_combine_tup(size_t & seed, const std::tuple<TupleTypes...> & tup) const
   {
     hash_combine(seed, std::get<Idx>(tup));
 
-    //  on to next element
-    hash_combine_tup<Idx + 1>(seed, tup);
+    if constexpr (Idx + 1< sizeof...(TupleTypes))
```

Comment:
@N-Dekker This was one of the most interesting replacements on a recursive variadic template function.

Reply:
@blowekamp Nice, but maybe it can even be further improved! Let me think...

Reply:
(hash_combine(seed, std::get<Idx>(tup), ...)

Reply:
Using a C++17 [fold expression](https://en.cppreference.com/w/cpp/language/fold) and [std::apply](https://en.cppreference.com/w/cpp/utility/apply)...?

```cpp
  operator()(std::tuple<TupleArgs...> tupleValue) const
  {
    size_t seed = 0;
    std::apply(
      [this, &seed](auto... tupleElement) { (hash_combine(seed, tupleElement), ...); },
      tupleValue);
    return seed;
  }
```

Disclaimer: uncompiled, untested code 🙀

Reply:
Yes, I just stumble onto std::apply now too. However, I think just: `std::apply(hash_combine,tupleValue);` is what is needed 🤔

Reply:
Thank you very much for this suggestion.